pkg/cluster/info.go

```diff
@@ -174,3 +174,16 @@ func (c *Info) OperatorNamespace() string {
 
 	return constants.OperatorNamespace
 }
+
+func (c *Info) GetBrokers(namespace string) (*v1alpha1.BrokerList, bool, error) {
+	brokerList := &v1alpha1.BrokerList{}
+	err := c.ClientProducer.ForGeneral().List(context.TODO(), brokerList, controllerClient.InNamespace(namespace))
+
+	if err == nil {
+		return brokerList, true, nil
+	} else if !apierrors.IsNotFound(err) && !meta.IsNoMatchError(err) {
+		return nil, false, errors.Wrapf(err, "error retrieving Broker in namespace %s", namespace)
+	}
+
+	return brokerList, false, nil
```

Comment:
The idiomatic pattern in Go is to look for non-nil `err`:

```suggestion
        if err != nil && !apierrors.IsNotFound(err) && !meta.IsNoMatchError(err) {
            return nil, errors.Wrapf(err, "error retrieving Broker in namespace %s", namespace)
        }
        
        return brokerList, nil
```

Reply:
Well, I just copied what is done for Submariner (https://github.com/submariner-io/subctl/blob/devel/pkg/cluster/info.go#L68-L72) and Service discovery.

Reply:
The usage is different: you’re returning in the `err == nil` branch, the code in `NewInfo` is filling in a value in the `err == nil` branch.